tilavarauspalvelu/integrations/email/sending.py

```diff
@@ -1,67 +1,52 @@
 from __future__ import annotations
 
 from copy import copy
+from dataclasses import asdict
 from itertools import batched
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, Any
 
 from django.conf import settings
 from django.core.mail import EmailMultiAlternatives
 
 from config.celery import app
+from tilavarauspalvelu.integrations.email.dataclasses import EmailData
 
 if TYPE_CHECKING:
     from collections.abc import Iterable
 
-    from tilavarauspalvelu.typing import EmailAttachment, EmailData
 
 __all__ = [
     "send_emails_in_batches_task",
+    "send_multiple_emails_in_batches_task",
 ]
 
 
 @app.task(name="send_emails_in_batches")
-def send_emails_in_batches_task(
-    *,
-    recipients: Iterable[str],
-    subject: str,
-    text_content: str,
-    html_content: str,
-    attachments: Iterable[EmailAttachment] = (),
-) -> None:
-    """
-    Sends an email message to in batches.
-
-    :param recipients: The recipients of the email.
-    :param subject: The subject of the email.
-    :param text_content: The text content of the email.
-    :param html_content: The HTML content of the email.
-    :param attachments: An attachments in the email.
-    """
+def send_emails_in_batches_task(email_data: EmailData | dict[str, Any]) -> None:
```

Comment:
I would remove `dict[str, Any]` from the typing so that when the function is used, typing would indicate that we should use the EmailData type instead of any dict being acceptable. I would then flip the guard clause to convert the celery-serialized dict to an `EmailData` instance, with a comment to explain it.